tap_github/streams.py

```diff
@@ -32,9 +32,11 @@ def path(self) -> str:  # type: ignore
         """Return the API endpoint path."""
         if "searches" in self.config:
             return "/search/repositories"
-        else:
+        elif "repositories" in self.config:
             # the `repo` and `org` args will be parsed from the partition's `context`
             return "/repos/{org}/{repo}"
+        else:
```

Comment:
Hi @ericboucher, that makes sense. Do you think we should place a safeguard in the config JSON schema itself? Something like

```json
{
  "$schema": "https://json-schema.org/draft-07/schema",
  "type": "object",
  "allOf": [
    {
      "type": "object",
      "properties": {
        "auth_token": {
          "type": "string"
        }
      }
    },
    {
      "oneOf": [
      	{
          "type": "object",
	      "properties": {
            "search": {
              "type": "string"
            }
          },
          "additionalProperties": false
        },
        {
          "type": "object",
          "properties": {
            "organizations": {
              "type": "array",
              "items": {
                "type": "string"
              }
            }
          },
          "additionalProperties": false
        },
        {
          "type": "object",
          "properties": {
            "organizations": {
              "type": "array",
              "items": {
                "type": "string"
              }
            }
          },
          "additionalProperties": false
        }
      ]
    }
  ]
}
```

Tested with https://www.jsonschemavalidator.net/, this would fail configurations which contain more than one from `{search, repositories, organizations}`

Reply:
I am not sure how you all do it in other taps. I think for our use-case, explicit documentation and a simple error as I proposed above should work fine. But if there is already a pattern used in other taps we could try to follow that.

Reply:
@aaronsteers Agreed. Got carried away by what would be possible with a JSON schema. Simple validation is perfect 👍